### PR TITLE
Open video for collaboration deep-links

### PR DESCRIPTION
**Pull request by MattIPv4**
## Describe your changes

To improve visibility and user experience, open the video popup by default when a deep-link anchor is used for a specific collaboration.

## Notes for testing your change

Valid deep-links open the popup reliably. Invalid deep-links do not.

### CODE REVIEW

**Comment by MattIPv4** [comment]
I'm not very happy with this approach currently, simulating clicks. Trying to figure out what methods are exposed for directly triggering lightbox.

File: apps/website/src/components/content/YouTube.tsx [hunk 1]
```diff
@@ -56,19 +58,35 @@ type TriggerProps = {
...
+    ({ videoId, caption, className, children }, ref) => {
+      // Expose a method to open the lightbox as the ref
+      const elm = useRef<HTMLAnchorElement>(null);
+      const open = useCallback(() => {
+        if (elm.current) elm.current.dispatchEvent(new Event("lightbox"));
+      }, []);
+      useEffect(() => {
+        if (typeof ref === "function") ref(open);
+        else if (ref) ref.current = open;
+      }, [ref, open]);
```
Comment (MattIPv4) [inline apps/website/src/components/content/YouTube.tsx:71]: My only question is whether it makes sense to expose a custom method here, or whether I should just expose the element and let the consumer dispatch the event?

**Comment by pjeweb** [comment]
I am not sure what i am missing, but would it not be easier to open the lightbox using the method provided by photoswipe? Thats what we are using for show and tell also. I drafted a quick alternative PR #429 

I'm not sure about the refs approach. I think having a method in refs is not as common and so its harder for people to get maybe, but the encapsulation does make some sense. Not sure. Maybe it would feel nice if you package the event trigger thingy in a separate hook and give it the element ref instead. Hope that makes sense. Hard to grok react if it gets this entangled.

**Review by pjeweb — APPROVED** [review]
Great PR 👏

**Pull request merged**